File: nodejs/src/models/payment.ts
import { Expose } from 'class-transformer';
import { IsOptional, IsPositive, IsString } from 'class-validator';

export class CreatePaymentBody {
  @IsOptional()
  payment_method: string;

  @IsPositive()
  amount: number;

  @IsString()
  currency: string;

  @IsOptional()
  @IsString()
  customer: string;

  @IsOptional()
  @IsString()
  complete_payment_url: string;

  @IsOptional()
  @IsString()
  error_payment_url: string;
}

export class PaymentResult {
  @Expose()
  @IsString()
  id: string;
}

File: python/src/models/payment.py
from typing import Optional
from pydantic import BaseModel


class CreatePaymentModel(BaseModel):
    amount: float
    currency: str
    payment_method: Optional[str]
    customer: Optional[str]
    complete_payment_url: Optional[str]
    error_payment_url: Optional[str]
   
    

class PaymentModel(BaseModel):
    id: str
    